src/zarr/abc/store.py

```diff
@@ -329,17 +329,17 @@ def supports_listing(self) -> bool:
         ...
 
     @abstractmethod
-    def list(self) -> AsyncGenerator[str, None]:
+    async def list(self) -> AsyncIterator[str]:
         """Retrieve all keys in the store.
 
         Returns
         -------
-        AsyncGenerator[str, None]
+        AsyncIterator[str, None]
         """
-        ...
+        return iter([])
```

Comment:
weird, you shouldn't need the implementation if you annotate with `abstractmethod`

Reply:
Indeed. I had to `yield` with `AsyncGenerator`. `AsyncIterator` has the additional benefit that an empty method is good enough.